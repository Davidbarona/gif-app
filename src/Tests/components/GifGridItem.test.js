import React from 'react'
import {GifGridItem} from '../../components/GifGridItem'
import {shallow} from 'enzyme'


describe('test on GifGridItem component',()=>{
    
    const title = "test title"
    const url ="https://localhost/somenthing.jpg"
    let wrapper = shallow(<GifGridItem title={title} url={url} />)


    test('must show a render component', () => { 
        
        expect(wrapper).toMatchSnapshot()
     })

     test('should to have a paragraph with the title', () => {
        const p =  wrapper.find('p')
        expect(p.text().trim()).toBe(title)
     } )

     test('should have the image equal to url and alt of the props', () => { 
         const img = wrapper.find('img')
         //console.log(img.prop('src'))
         expect(img.prop('src')).toBe(url)
         expect(img.prop('alt')).toBe(title)
      })

    test('should have animate__rubberBand ', () => {
        const div = wrapper.find('div')
        const className = div.prop('className')
        // expect(div.prop('className')).toBe('card animate__animated animate__rubberBand')
        expect(className.includes('animate__rubberBand')).toBe(true)

     })
    
    

})

