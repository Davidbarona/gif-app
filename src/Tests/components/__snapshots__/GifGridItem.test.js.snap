// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifGridItem component must show a render component 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="test title"
    src="https://localhost/somenthing.jpg"
  />
  <p>
    test title
  </p>
</div>
`;
